src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/VisualStudioBrowserRefreshServer.cs

```diff
@@ -0,0 +1,104 @@
+
+    private static string GetMiddlewareAssemblyPath()
+        => ProjectHotReloadSession.GetInjectedAssemblyPath(MiddlewareTargetFramework, "Microsoft.AspNetCore.Watch.BrowserRefresh");
+
+    protected override bool SuppressTimeouts
+        => false;
+
+    // for testing
+    internal Task? WebSocketListeningTask { get; private set; }
+
+    protected override ValueTask<WebServerHost> CreateAndStartHostAsync(CancellationToken cancellationToken)
+    {
+        var httpListener = CreateListener(projectName, port, sslPort);
+        WebSocketListeningTask = ListenAsync(cancellationToken);
+
+        return new(new WebServerHost(httpListener, GetWebSocketUrls(projectName, port, sslPort), virtualDirectory));
+
+        async Task ListenAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                httpListener.Start();
+
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    Logger.LogDebug("Waiting for a browser connection");
+
+                    // wait for incoming request:
+                    var context = await httpListener.GetContextAsync();
+                    if (!context.Request.IsWebSocketRequest)
+                    {
+                        context.Response.StatusCode = 400;
+                        context.Response.Close();
+                        continue;
+                    }
+
+                    try
+                    {
+                        // Accepting Socket Next request. If the context has a "Sec-WebSocket-Protocol" header it passes back in the AcceptWebSocket
+                        var protocol = context.Request.Headers["Sec-WebSocket-Protocol"];
+                        var webSocketContext = await context.AcceptWebSocketAsync(subProtocol: protocol).WithCancellation(cancellationToken);
+
+                        _ = OnBrowserConnected(webSocketContext.WebSocket, webSocketContext.SecWebSocketProtocols.FirstOrDefault());
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.LogError("Accepting web socket exception: {Message}", e.Message);
+
+                        context.Response.StatusCode = 500;
+                        context.Response.Close();
+                    }
+                }
+            }
+            catch (OperationCanceledException)
+            {
+                // nop
+            }
+            catch (Exception e)
+            {
+                Logger.LogError("HttpListener exception: {Message}", e.Message);
+            }
+        }
+    }
+
+    private static HttpListener CreateListener(string projectName, int port, int sslPort)
+    {
+        var httpListener = new HttpListener();
+
+        httpListener.Prefixes.Add($"https://localhost:{sslPort}/{projectName}/");
+        if (sslPort >= 0)
+        {
+            httpListener.Prefixes.Add($"http://localhost:{port}/{projectName}/");
+        }
```

Comment:
Is this backwards? Should it be:


```suggestion
        httpListener.Prefixes.Add($"http://localhost:{port}/{projectName}/");
        if (sslPort >= 0)
        {
            httpListener.Prefixes.Add($"https://localhost:{sslPort}/{projectName}/");
        }
```

Reply:
Yes! Good catch.